### PR TITLE
chore: move x64 CI to GHA

### PR DESCRIPTION
**Pull request by mayeut**
Travis CI does not offer free x64 CI for OpenSource projects anymore. Move to GitHub Actions for x64 tests.

Sample build is there: https://github.com/mayeut/base64/actions/runs/2323476527

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Test
+
+on:
+  push:
+  pull_request:
```
Comment (aklomp) [inline .github/workflows/test.yml:5]: I think this can be an array?

```yaml
on: [push, pull_request]
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            compiler: "clang"
+            openmp: "0"
+    env:
+      OPENMP: ${{ matrix.openmp }}
+      OMP_NUM_THREADS: ${{ ((matrix.openmp == 1) && '4') || '0' }}
+      CC: ${{ matrix.compiler }}
+      OBJCOPY: ${{ (startsWith(matrix.runner, 'macos') && 'echo') || 'objcopy' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (aklomp) [inline .github/workflows/test.yml:28]: This one's already at `v3`.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    strategy:
+      fail-fast: false
+      matrix:
+        runner: ["ubuntu-20.04"]
+        compiler: ["gcc", "clang"]
+        openmp: ["0", "1"]
+        include:
+          - runner: "macos-11"
+            compiler: "clang"
+            openmp: "0"
```
Comment (aklomp) [inline .github/workflows/test.yml:20]: I believe GitHub also has Windows runners available? Well, those could be added later, no action needed.

File: test/ci/amd64.sh [hunk 4]
```diff
@@ -5,10 +5,17 @@ export SSSE3_CFLAGS=-mssse3
 export SSE41_CFLAGS=-msse4.1
 export SSE42_CFLAGS=-msse4.2
 export AVX_CFLAGS=-mavx
-export AVX2_CFLAGS=-mavx2
+# no AVX2 on GHA macOS
+if [ "${OBJCOPY:-}" != "echo" ]; then
```
Comment (aklomp) [inline test/ci/amd64.sh:9]: This is a bit of a hack. It's probably nicer to do OS detection than to rely on `OBJCOPY` being set to the magic string `"echo"`.

File: .github/workflows/test.yml (hunk 2 above)
Comment (mayeut) [reply .github/workflows/test.yml]: done.

File: .github/workflows/test.yml (hunk 1 above)
Comment (mayeut) [reply .github/workflows/test.yml]: done

File: .github/workflows/test.yml (hunk 3 above)
Comment (mayeut) [reply .github/workflows/test.yml:18]: windows now handled with cmake.

File: test/ci/amd64.sh (hunk 4 above)
Comment (mayeut) [reply test/ci/amd64.sh]: done

**Comment by mayeut** [comment]
Tests with CMake have been added which showed an issue with macOS build.

**Pull request closed without merging**